### DISCUSSION TITLE
API stability?

### DISCUSSION THREAD

**Discussion by AshleyScirra** [gpuweb/gpuweb#2671] [Q&A] [answered]
There have been a spate of breaking changes lately, both in WebGPU and WGSL. That's fine for experimentation purposes. But we'd like to start experimenting more widely with users and third-party developers who need to port their own shaders to WGSL as well. It would be good to have some degree of API stability before doing this, to minimise the risk that things break while we're experimenting more widely.

Chrome's roadmap appears to show WebGPU shipping as soon as May in Chrome 102, which would suggest that we're approaching some degree of stability. It would be useful though to have some more official confirmation: are any more breaking API changes expected before MVP? Will there be some kind of "stable spec" announcement at some point?

**Comment by kevmoo** [comment]
FYI: https://chromestatus.com/feature/6213121689518080 – the 102 expected release has been removed from the Chrome tracker.

**Comment by kainino0x** [comment, marked as answer]
A few more breaking changes are still expected. The API changes will continue to be nonstructural, and WGSL is reaching stability but I'm not 100% sure how many more changes are expected.

There will be a stable spec announcement, and Chrome won't ship before we reach that point (we intend for them to coincide). At @kevmoo's prompting I've removed the 102 milestone from chromestatus. We're still figuring out what to put there, but ~tentatively expecting somewhere around 104-105.~

**Comment by lisanlow** [comment]
@kainino0x May I confirm with you that WebGPU would be shipping as of build 105?  Any updates or news on WebGPU shipping date?

**Reply by Kangz** [reply]
WebGPU will not be shipping in Chrome 105. There is still the Origin Trial though. Breaking changes are few and far between now but we can't ship yet because there are still topics to be discussed.
